File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums1 = {}
        for i in range(len(nums)):
            nums1[nums[i]] = i
        
        for i in range(len(nums)):
            if target - nums[i] in nums1 and i != nums1[target - nums[i]]:
                return [i, nums1[target - nums[i]]]
        

File: 1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        
        var map = [Int: Int]()

        for i in 0..<nums.count {
            if let index = map[target - nums[i]] {
                return [i, index]
            }

            map[nums[i]] = i
        }

        return []
    }
}